tests/test_workspaces.py

```diff
@@ -60,4 +68,57 @@ def test_number_of_existing_projects(self, create_driver, prep_properties):
         pp = ProjectsPage()
         number_of_displayed_projects = pp.get_projects_number_in_page()
         number_of_projects_in_workspace = pp.get_projects_number_from_workspace()
-        assert_that(number_of_displayed_projects).is_equal_to(number_of_projects_in_workspace)
\ No newline at end of file
+        assert_that(number_of_displayed_projects).is_equal_to(number_of_projects_in_workspace)
+
+    @allure.description("Selecting and adding a project to workspace")
+    @pytest.mark.run(order=5)
+    def test_add_project_to_workspace(self, create_driver, prep_properties):
+        self.login(prep_properties)
+        pp = ProjectsPage()
+        ptp = ProjectTypePage()
+        tp = TemplatesPage()
+        pep = ProjectEditPage()
+        before = pp.get_projects_number_in_page()
+        pp.create_new_project()
+        ptp.select_project(self._project_type)
+        tp.choose_template(self._template_type)
+        pep.edit_project_prep(self._project_name, self._final_slide)
+        pep.click_save_and_exit()
+        after = pp.get_projects_number_in_page()
+        assert_that(after).is_equal_to(before + 1)
+
+    @allure.description("Search for an existing project")
+    @pytest.mark.run(order=6)
+    def test_search_project(self, create_driver, prep_properties):
+        self.login(prep_properties)
+        pp = ProjectsPage()
+        pp.search_project(self._project_name)
+        assert_that(pp.is_project_found(self._project_name)).is_true()
+
+    @allure.description("Search for a non existing project")
+    @pytest.mark.run(order=7)
+    def test_search_for_non_existing_project(self, create_driver, prep_properties):
+        self.login(prep_properties)
+        pp = ProjectsPage()
+        pp.search_project("Non Existing")
+        assert_that(pp.get_no_project_found_msg()).is_equal_to(self._no_project_found_msg)
+
+    @allure.description("Cancel project deletion")
+    @pytest.mark.run(order=8)
+    def test_cancel_project_deletion(self, create_driver, prep_properties):
+        self.login(prep_properties)
+        pp = ProjectsPage()
+        before = pp.get_projects_number_in_page()
+        pp.delete_project(self._project_name, "cancel")
+        after = pp.get_projects_number_in_page()
+        assert_that(after).is_equal_to(before)
+
+    @allure.description("Deleting an existing project from workspace")
+    @pytest.mark.run(order=9)
+    def test_delete_project(self, create_driver, prep_properties):
+        self.login(prep_properties)
+        pp = ProjectsPage()
```

Comment:
we should us the real name of the class and not pp  when init pages - please fix for all project

Reply:
This is how Gal taught me to write instances of a class in test methods :D
However, I will change them.